### PR TITLE
[Cosmetic] Refactorings & Fantomas integration

### PR DESCRIPTION
**Pull request by CaptnCodr**
This PR includes:

- Resolved warnings: especially type warnings `list<_>`, `array<_>`, `seq<_>` were removed twice, it runs under `IEnumerable` too
    - `CollectionContainsConstraint` was marked as obsolete: it's now `Has.Some.EqualTo` as recommended
- Several indentation alignments in `CustomMatchers.fs`: 4x tabs (4 spaces) in function body
- Removed redundant code: `new Type`, `open NHamcrest.Core`, `yield`s in `seq { yield ... }`
    - cut down usings `System.Collections.IEnumerable` but `System.Collections` were opened
- Follow style guidelines especially type declarations `a: Type` ([docs.microsoft](https://docs.microsoft.com/en-us/dotnet/fsharp/style-guide/formatting))
- Old comments they were never touched

Altogether, the code is more readable.

### CODE REVIEW

**Comment by sergey-tihon** [comment]
With such big reformatting, I believe it make sense to adopt https://github.com/fsprojects/fantomas
Slightly tune defaults to align with this project code style [(more)](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#using-the-command-line-tool) and format code on each build from FAKE

@CaptnCodr do you want to try this?

**Comment by CaptnCodr** [comment]
I didn't ever use Fantomas but I'll try this.

**Comment by sergey-tihon** [comment]
Thank you!  I also never integrate it with repo before.
Here you can find great talk from Florian with intro to Fantomas and live demo how to integrate it into new repo and configure to minimize PR diff (align with project styles) https://youtu.be/ybkYHYKYeNw?t=4467

**Comment by CaptnCodr** [comment]
It does make sense to push the fantomas integration into this PR, isn't it?

**Comment by sergey-tihon** [comment]
Yes, I also think so

**Comment by CaptnCodr** [comment]
Hey @sergey-tihon, I added Fantomas now. :)
I just use the Default config `FormatConfig.Default` plus only the code in `src` because there is `ofCaseTests.fs` which will be formatted on every build. I don't know why. I exclude the `AssemblyInfo.fs` because this file is autogenerated.

**Comment by sergey-tihon** [comment]
I am not really have with formatting like 
image

I've upgraded Fantomas to latest prerelease version and slightly upgrade config.
What do you think about current format?

**Comment by CaptnCodr** [comment]
Oh yes. That's definitely better.

**Comment by CaptnCodr** [comment]
We could replace all `.` in test names by `[dot]` and e.g. `1 .. 10` by `1 to 10` to have a better structural overview in VS.
Additionally, we could ship this PR with PR #147 in FsUnit v4.0? :)

File: src/FsUnit.NUnit/Equality.fs [hunk 1]
```diff
@@ -7,19 +7,35 @@ open LanguagePrimitives
...
-        |_ ->
-            Is.EqualTo(x)
-    static member IsNotEqualTo(x : 'T) = NotConstraint(Equality.IsEqualTo(x))
+type Equality<'T when 'T: equality> =
+    static member Reference =
+        System.Func<'T, 'T, bool>(fun x y ->
+            let f =
+                FastGenericEqualityComparer<'T>.Equals(x, y)
+
+            printfn "Ref returns %b" f
```
Comment (sergey-tihon) [inline src/FsUnit.NUnit/Equality.fs:16]: @marklam I missed this line in your formatting. Why do you `printf` to console here?

**Pull request merged**